src/server/server_state.cc

```diff
@@ -262,8 +265,9 @@ void ServerState::ConnectionsWatcherFb(util::ListenerInterface* main) {
         is_replica = dfly_conn->cntx()->replica_conn;
       }
 
-      if ((phase == Phase::READ_SOCKET || dfly_conn->IsSending()) && !is_replica &&
-          dfly_conn->idle_time() > timeout) {
+      if ((timeout != 0 && phase == Phase::READ_SOCKET && !is_replica &&
```

Comment:
please factor out into 2 boolean variables for clarity: `idle_read` and `stuck_sending`

btw, I think for `stuck_sending` you also need `!is_replica` because we have a different setting to break replication connections.